### PR TITLE
HIPP-1293: Domain and subdomain on generate

### PR DESCRIPTION
**Pull request by llmikeyj**

### CODE REVIEW

File: test/uk/gov/hmrc/apihubapplications/controllers/DeploymentsControllerSpec.scala [hunk 1]
```diff
@@ -166,6 +172,33 @@ class DeploymentsControllerSpec
       }
     }
 
+    "must return 400 Bad Request when subdomain is present but domain is not" in {
```
Comment (PaulCDurham) [inline test/uk/gov/hmrc/apihubapplications/controllers/DeploymentsControllerSpec.scala:175]: As per above, I don't think we should do this.

File: test/uk/gov/hmrc/apihubapplications/controllers/DeploymentsControllerSpec.scala [hunk 2]
```diff
@@ -65,7 +65,9 @@ class DeploymentsControllerSpec
           "teamId",
           "oas",
           false,
-          "status"
+          "status",
+          None,
+          None
```
Comment (PaulCDurham) [inline test/uk/gov/hmrc/apihubapplications/controllers/DeploymentsControllerSpec.scala:70]: We're using None everywhere. Could we use Some(stuff) in maybe just the first positive test at least?

**Review by PaulCDurham — CHANGES_REQUESTED** [review]
I don't think we should do that validation in the backend - see below. Happy to discuss as that's just an opinion.

File: app/uk/gov/hmrc/apihubapplications/controllers/DeploymentsController.scala [hunk 3]
```diff
@@ -42,6 +42,9 @@ class DeploymentsController @Inject()(
     implicit request =>
       val jsReq = request.body
       jsReq.validate[DeploymentsRequest] match {
+        case JsSuccess(deploymentsRequest, _) if deploymentsRequest.subdomain.isDefined && deploymentsRequest.domain.isEmpty =>
```
[the remark this answers is not in the document]
Comment (llmikeyj) [reply app/uk/gov/hmrc/apihubapplications/controllers/DeploymentsController.scala]: ok

File: test/uk/gov/hmrc/apihubapplications/controllers/DeploymentsControllerSpec.scala (hunk 2 above)
Comment (llmikeyj) [reply test/uk/gov/hmrc/apihubapplications/controllers/DeploymentsControllerSpec.scala]: ok

File: test/uk/gov/hmrc/apihubapplications/controllers/DeploymentsControllerSpec.scala (hunk 1 above)
Comment (llmikeyj) [reply test/uk/gov/hmrc/apihubapplications/controllers/DeploymentsControllerSpec.scala]: ok

**Review by PaulCDurham — APPROVED** [review]

**Pull request merged**